src/main/resources/api/openapi.yaml

```diff
@@ -3035,6 +3171,43 @@ paths:
               schema:
                 $ref: '#/components/schemas/ApiError'
 
+  /wallet/transactionsByScanId:
+    parameters:
+      - in: query
+        name: scanId
+        required: true
+        description: Scan id
+        schema:
+          type: string
+    get:
+      security:
+        - ApiKeyAuth: [ api_key ]
+      summary: Get scan-related transactions by scan id
+      operationId: walletTransactionsByScanId
+      tags:
+        - wallet
+      responses:
+        '200':
+          description: Scan-related transactions
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/WalletTransaction'
+        '404':
+          description: Transactions with related scan id not found in wallet
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ApiError'
+        default:
+          description: Error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ApiError'
```

Comment:
For now, I think it is not critical, because blocks don't contain so many transactions, so it could be implemented
later:    https://github.com/ergoplatform/ergo/issues/1271

Reply:
Please also note that calling allWalletTxs() to get scans of one scan could be very inefficient, consider to add a scan <-> transaction index, like done for boxes already.